### DISCUSSION TITLE
텃밭 테이블의 컬럼에 study_id를 추가하자는 의견

### DISCUSSION THREAD

**Discussion by lcqff** [BDD-CLUB/01-doo-re-back#163] [논의 사항]
```sql
create table garden
(
    id               bigint auto_increment primary key,
    contributed_date date         not null,
    type             varchar(255) not null,
    is_deleted       boolean      not null,
    contribution_id  bigint       not null,
    team_id          bigint       not null,
    member_id        bigint       not null,
    created_at       datetime(6),
    updated_at       datetime(6)
);
``` 

현재 텃밭 테이블의 필드는 위와 같습니다! 

텃밭 테이블은 스터디/팀에 학습자료가 업로듸 되거나 스터디의 커리큘럼이 체크될때마다 생성됩니다. 
현재는 팀의 텃밭 모델을 생성하고 일정 기간동안의 텃밭 데이터 개수를 총합하여 팀의 랭크를 구하는데 쓰이고 있습니다. 

그러나 스터디 목록 조회를 개발하며 스터디 목록또한 랭킹순으로 정렬하는 편이 좋을 거 같다는 생각이 들었습니다.
랭킹을 단순 스터디 진행률로 계산하면 편하겠지만 그보다는 이번주동안 이 스터디의 기여도를 계산하는 편이 좋겠다 생각합니다. (차후 BDD가 프로젝트에서 스터디로 방향성을 바꾸며, 두레를 스터디 관리용으로 사용할 가능성을 생각해보면 더욱)

이때 텃밭을 통해 스터디 기여도를 계산할 방법이 없습니다 (텃밭 데이터에서 이 데이터는 어느 스터디에서 왔는지 출처를 찾을 수 없습니다)

두가지 방법이 있습니다

**1. 그냥 커리큘럼 개수 + 업로드된 자료 계수를 직접 계산한다.
2. 텃밭 데이터에 study_id, is_from_study(is_team_contribution? 아무튼 스터디에서 기여한 데이터가 아니라 팀에서 기여된 데이터 (ex 팀에서 업로드한 학습자료) 컬럼 두개를 추가하여 garden에서 데이터를 가져온다.**

솔직히 지금 상황만 따지면 1번도 괜찮긴 하나, 차후 스터디에서의 기여 방법이 curriculum check, 학습자료 업로드 외로 더 생길 경우 코드를 수정해야 하기에 2번 쪽을 생각하고 있습니다.

다른 분들의 의견이 궁금합니다!

**Comment by JJimini** [comment]
우선 스터디 기여도라는 것은 스터디가 팀에 가한 기여도라고 이해했습니다.

하지만 해당 방법은 스터디 참여인원에 따라 랭킹이 결정되는 문제가 발생할 수 있을 것 같습니다. (절대 기준 계산 시)

이 문제를 해결하기 위해 상대 기준 계산을 하는 방법이 있을 것 같은데, 이 계산법과 기준에 대해서는 기획적인 부분이라고 생각하며, 백엔드에서 결정할 부분이 아닌 모두의 의견을 듣고 결정해야하는 문제라고 생각합니다.

**Comment by lcqff** [comment]
그렇네요 절대 기준으로 계산시 스터디 참여인원에 따른 랭킹 문제가 있을거란 생각은 미쳐못했습니다!  
저도 상대기준 계산이 좋을거 같습니다. 팀원들과 상의한 뒤에 결정하는걸로 합시다! (+팀 랭킹도 상대기준으로 계산할지 얘기해봅시다)

그러면 혹시 텃밭 테이블의 컬럼에 study_id를 추가하자는 부분은 동의가 된 사항일까요?👀 
동의가 됐다면 다음주까지 스터디 목록 조회의 개발은 진행해야 하는지라, 일단 텃밭 테이블의 데이터를 상대 기준으로 계산한 다음 상의후 결정된 방향대로 수정해도 될까요? 절대 <-> 상대 계산으로 바꾸는 거 자체는 간단한 계산이니까요 (절대/상대가 아니라 다른 의견이 나올수도 있겠지만...)

**Reply by JJimini** [reply]
큰 상관은 없을 것 같지만, 지금은 현행 유지하는 것이 더 좋지 않을까요?? 
만들어 놓고 지우는 것보단, 확실하게 의견을 나눈 후에 만들면 일을 2번 할 필요가 없으니까요!
그리고 스터디를 순위를 매기는 것도 이번 주 (배타 버전 이전)에 할 만큼 급한 issue가 아닌 것 같아서요...!

**Reply by lcqff** [reply]
좋습니다~ 그러면 랭킹 정렬 기능을 제외한 스터디 목록 조회만 구현하도록 하고 이후에 랭킹 정렬 기능을 추가하는 걸로 하죠!

**Comment by lcqff** [comment]
논의 결과 Team Ranking과 정렬 규칙이 달라지는 사용자 입장에서 헷갈릴 수 있고, 팀 인원이 많은 것이 높은 랭킹에 도움이 되긴 하나 그만큼 스터디 진행도 힘들것이라는 의견이 있어 Team Ranking과  통일하기로 결정됐습니다. (2번 방식)
